### PR TITLE
Cover Image Dag

### PR DESCRIPTION
**Pull request by vinnyjth**
@tomthornton this PR is a bit weird, it's a "revert" of a "revert" but it's got all your changes

### CODE REVIEW

File: dags/rock_cover_image.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            parents = requests.get(
+                    f"{Variable.get(kwargs['client'] + '_rock_api')}/ContentChannelItemAssociations",
+                    params={
+                        "$filter": f'ChildContentChannelItemId eq {contentItemId}',
+                    },
+                    headers=headers).json()
+
+            if(len(parents) == 1):
+                parentContentItemId = parents[0]['ContentChannelItemId']
+                parentContentItem = requests.get(
```
Comment (vinnyjth) [inline dags/rock_cover_image.py:67]: Your approach here looks great! I am a bit worried about the number of queries we'll have to make to Rock per-content item. We have all the data we need in Postgres _except_ the `key` :/ What do you think about moving that key into postgres long term? Or maybe we split shoveling into two stages 

1.  rock -> postgres, we don't look at parents 
2. postgres -> postgres, we look for all items that don't have cover images and give them their parents cover image, purely by querying postgres.

File: dags/rock_cover_image.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                    "ContentItemId": content_item['Id'],
+                    "CoverImageId": coverImageId
+                }
+
+
+    def update_content_item_cover_image(args):
+        if(args):
+            contentItemId = args['ContentItemId']
+            coverImageId = args['CoverImageId']
+            return pg_hook.run('UPDATE "contentItems" SET "coverImageId" = %s WHERE "originId"::Integer = %s', True, (coverImageId, contentItemId))
```
Comment (vinnyjth) [inline dags/rock_cover_image.py:83]: we talked about this in slack, but we should be setting `coverImageId` to the `media.id`

Comment (vinnyjth) [reply dags/rock_cover_image.py:83]: https://github.com/ApollosProject/apollos-shovel/pull/7/files#diff-8b958333e85b50db625a8a72e3523e1ff545dd54cc97728257a758b9ae0c80f2R162-R166

Look here for an example of how you can still use the `originId` to set the `mediaId`

File: dags/rock_cover_image.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+def get_images (content_item):
+    def filter_images( attribute ):
+            attributeKey = attribute['Key']
+            attributeValue = content_item['AttributeValues'][attributeKey]['Value']
+
+            return attribute['FieldTypeId'] == 10 or ('image' in attributeKey.lower() and isinstance(attributeValue, str) and attributeValue.startswith('http'))
+
+    return list(filter(filter_images, content_item['Attributes'].values()))
+
+def get_best_image_id (images):
```
Comment (vinnyjth) [inline dags/rock_cover_image.py:14]: Nice work! you captured this logic well

**Comment by vinnyjth** [comment]
@tomthornton Left some feedback above :)

File: dags/rock_cover_image.py (hunk 2 above)
Comment (tomthornton) [reply dags/rock_cover_image.py]: Yep! just added a commit for this

File: dags/rock_cover_image.py (hunk 3 above)
Comment (tomthornton) [reply dags/rock_cover_image.py:14]: Thanks!

**Comment by tomthornton** [comment]
@vinnyjth updated to grab the right Id. As far as storing "key" long term goes, I think that would be fine but wouldn't it be better to store Parent -> Child relationships of the content items instead? I feel like that will be more useful in the long term.

**Comment by vinnyjth** [comment]
@tomthornton What do you think about commenting out the "pull image from parent" logic for now, merging this, and then seeing if we can come up with a sql query solution to solve the parent logic?

**Comment by tomthornton** [comment]
@vinnyjth Sounds good to me. Shouldn't be too hard.

**Pull request merged**